hs_core/hydroshare/utils.py

```diff
@@ -121,6 +124,143 @@ def group_from_id(grp):
     return tgt
 
 
+def get_user_zone_status_info(user):
+    """
+    This function should be called to determine whether the site is in production and whether user zone
+    functionality should be enabled or not on the web site front end
+    Args:
+        user: the requesting user
+    Returns:
+        in_production, enable_user_zone where both are boolean indicating whether the site is
+        in production and whether user zone functionality should be enabled or not on the web site front end
+    """
+    if user is None:
+        return None, None
+    if not hasattr(user, 'userprofile') or user.userprofile is None:
+        return None, None
+
+
+    in_production = True if settings.IRODS_USERNAME == settings.HS_WWW_IRODS_PROXY_USER else False
+    enable_user_zone = user.userprofile.create_irods_user_account
+    if not in_production and enable_user_zone:
+        # if these settings are not empty, for example, in users' local
+        # development environment for testing, user_zone selection is shown
+        if (not settings.HS_WWW_IRODS_PROXY_USER_PWD or \
+            not settings.HS_WWW_IRODS_HOST or not settings.HS_WWW_IRODS_ZONE):
+            enable_user_zone = False
+    return in_production, enable_user_zone
+
+
+def is_federated(homepath):
+    """
+    Get size of a data object from iRODS user zone
```

Comment:
@pkdash Thanks for catching this. Will update the docstring to reflect its true functionality.